### PR TITLE
Add new package: jetty-project

### PR DESCRIPTION
**Pull request by darmac**

### CODE REVIEW

**Review by adamjstewart — APPROVED** [review]

File: var/spack/repos/builtin/packages/jetty-project/package.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+class JettyProject(MavenPackage):
+    """Jetty is a lightweight highly scalable java based web
+    server and servlet engine."""
+
+    homepage = "https://www.eclipse.org/jetty"
+    url      = "https://github.com/eclipse/jetty.project/archive/jetty-9.4.31.v20200723.tar.gz"
+
+    version('9.4.31.v20200723', sha256='3cab80ddc14763764509552d79d5f1f17b565a3eb0a1951991d4a6fcfee9b4b1')
+    version('9.4.30.v20200611', sha256='fac8bb95f8e8de245b284d359607b414893992ebb4e2b6e3ee40161297ea2111')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/jetty-project/package.py:17]: Add this:
```python
depends_on('java@8:', type=('build', 'run'))
depends_on('maven@3:', type='build')
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/jetty-project/package.py (hunk 1 above)
Comment (darmac) [reply var/spack/repos/builtin/packages/jetty-project/package.py:17]: Is there anyway to get the version constraintion info directly?
Or we should check official web to get these info?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/jetty-project/package.py:17]: I get it from the pom.xml file. See the docs I wrote on the MavenPackage build system.

Comment (darmac) [reply var/spack/repos/builtin/packages/jetty-project/package.py:17]: Refined it in https://github.com/spack/spack/pull/18555/commits/af774a9152323edfcbb23b2e3212e6f9584237bc

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/jetty-project/package.py:17]: Where did you get 3.6 from? Do newer versions of Java not work?

Comment (darmac) [reply var/spack/repos/builtin/packages/jetty-project/package.py:17]: hmm, I checked that with a wrong position:
```
  35     <junit.version>5.6.2</junit.version>
  36     <maven.version>3.6.0</maven.version>
  37     <maven.resolver.version>1.3.1</maven.resolver.version>
```

I should fix it by refer here:
```
 156                 <requireMavenVersion>
 157                   <version>[3.0.0,)</version>
 158                   <message>[ERROR] OLD MAVEN [${maven.version}] in use, Jetty ${project.version} requires Maven 3.0.0 or newer</message>
 159                 </requireMavenVersion>

```

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**